virttest/utils_test/__init__.py

```diff
@@ -838,205 +848,284 @@ def transfer_data(session, host_cmd, guest_cmd, n_time, timeout,
-               'package': [],
-               'git': ['varianter_yaml_to_mux', 'html']}
-    prerequisites = ['python', 'git']
-
-    def env_check():
+    def manage_session(self):
+        try:
+            if self.vm or self.remote_host:
+                self.session = self.get_session()
+            return func(self)
+        finally:
+            if self.vm or self.remote_host:
+                self.session.close()
+    return manage_session
+
+
+class AvocadoGuest(object):
+
+    def __init__(self, vm, params, test, testlist=[], timeout=3600,
+                 testrepo="https://github.com/avocado-framework-tests/avocado-misc-tests.git",
+                 installtype="pip", avocado_vt=False, reinstall=False,
+                 add_args="", ignore_result=True):
+        """
+        Class to run Avocado/Avocado-VT tests inside guest
+
+        :param vm: VM object
+        :param params: VM param
+        :param test: test object
+        :param testlist: testlist as list of tuples like (testcase, muxfile)
+        :param timeout: test timeout
+        :param testrepo: test repository default is avocado-misc-tests
+        :param installtype: how to install avocado, supported types
+                            pip, package, git
+        :param reinstall: flag to reinstall incase of avocado present inside guest
+        :param add_args: additional arguments to be passed to the avocado cmdline
+        :param ignore_result: True or False
+        :return: Bool result status
+        """
+        self.vm = vm
+        self.params = params
+        self.test = test
+        self.testlist = testlist
+        self.timeout = timeout
+        self.test_repo = testrepo
+        self.installtype = installtype
+        self.avocado_vt = avocado_vt
+        self.reinstall = reinstall
+        self.add_args = add_args
+        self.ignore_result = "yes" if ignore_result else "no"
+        self.session = None
+        self.test_path = self.params.get("vm_test_path", "/var/tmp/avocado/")
+        self.result_path = os.path.join(self.test_path, "results")
+        self.avocado_repo = self.params.get("avocado_repo_path",
+                                            "https://github.com/avocado-framework/avocado.git")
+        self.plugins = {'pip': ['avocado-framework-plugin-varianter-yaml-to-mux', 'avocado-framework-plugin-result-html'],
+                        'package': [],
+                        'git': ['varianter_yaml_to_mux', 'html']}
+        self.prerequisites = {'git': ['git'], 'python': ['python', 'python-pip'], 'python2': ['python2', 'python2-pip']}
+        if self.avocado_vt:
+            self.vt_type = self.params.get("vt_type", "libvirt")
+            self.vt_only_filter = self.params.get("vt_only_filter", "")
+            self.vt_no_filter = self.params.get("vt_no_filter", "")
+            self.guest_image = self.params.get("vt_guest_image", "")
```

Comment:
prefix all params with test specific tags as this confuse with L0 params...
e-g:- `avocadoguesttest_vt_type`

Reply:
right, I am handling it in different way for differentiating with Level and guest specific params.